### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by BogdanScherban**
Logout process:

1) User click on logout button;
2) Run api request to api/logout
3) Server should return request with redirectURL
4) Clean token from COOKIE and remove userId from localStorage
5) Redirect to URL from request

### CODE REVIEW

File: src/core/dataProviders/authProvider.js [hunk 1]
```diff
@@ -35,13 +35,28 @@ const FetchLogin = (resolve, reject) => {
         });
 };
 
+const FetchLogout = (resolve, reject) => {
```
Comment (fzaninotto) [inline src/core/dataProviders/authProvider.js]: I find it weird to extract the Promise callback like that. It makes your code harder to follow IMO.

Why do you need an additional Promise when you can write it all inline line 51:

```js
if (localStorage.getItem('userId') && token) {
    fetch(`${domainName}/api/logout`, options)
        .then(res => res.json())
        .then(response => {
            document.cookie = 'JSESSIONID=;';
            localStorage.removeItem('userId');
            localStorage.removeItem('username');
            localStorage.removeItem('role');
            window.location = response.redirectUrl;
         });
```

File: src/core/dataProviders/authProvider.js [hunk 2]
```diff
@@ -35,13 +35,28 @@ const FetchLogin = (resolve, reject) => {
...
-        localStorage.removeItem('given_name');
-        localStorage.removeItem('family_name');
-        localStorage.removeItem('role');
+        if (localStorage.getItem('userId') && token) {
+            new Promise(FetchLogout).then(redirectUrl => {
+                document.cookie = 'JSESSIONID=;';
+                localStorage.removeItem('userId');
+                localStorage.removeItem('username');
+                localStorage.removeItem('role');
+                window.location = redirectUrl;
```
Comment (fzaninotto) [inline src/core/dataProviders/authProvider.js]: react-admin already does a redirection to the login page *before* calling the auth provider. Besides, I find it weird that you trust the API to tell you where you should redirect the user - that's the frontend's business. So you should not change the window location.

File: src/core/dataProviders/authProvider.js (hunk 1 above)
Comment (BogdanScherban) [reply src/core/dataProviders/authProvider.js]: Fixed. I removed promise ad changes code by this way.

File: src/core/dataProviders/authProvider.js (hunk 2 above)
Comment (BogdanScherban) [reply src/core/dataProviders/authProvider.js]: @fzaninotto 
Yes, RA provides redirect to login page - it works!
But we need in redirect to OpenID login page. There are URL something like:
https://{domain name}/openid/interaction/3eb0f3d9-c23c-4482-8e27-eed8c07e7575 

We have domain name at frontend part, but where can we get this code: 3eb0f3d9-c23c-4482-8e27-eed8c07e7575 ? Besides this, this code is differ each time.

For this reason we use redirect URL from response with window.location.

**Comment by BogdanScherban** [comment]
@fzaninotto 
I fixed first note and added my explanation for the second one.

File: src/core/dataProviders/authProvider.js (hunk 2 above)
Comment (fzaninotto) [reply src/core/dataProviders/authProvider.js]: Right, I understand.

**Pull request merged**